src/app/phast/losses/operations/co2-savings-phast/co2-savings-phast.component.ts

```diff
@@ -0,0 +1,362 @@
+  }
+
+  setZipcode() {
+    this.form.controls.userEnteredBaselineEmissions.patchValue(false);
+    this.setUserEnteredModificationEmissions(false);
+    this.setSubRegionData();
+  }
+
+  setUserEnteredModificationEmissions(isUserEnteredValue: boolean) {
+    let modificationCo2SavingsData: Co2SavingsData = this.phastCO2SavingsService.modificationCo2SavingsData.getValue();
+    if (modificationCo2SavingsData) {
+      modificationCo2SavingsData.userEnteredModificationEmissions = isUserEnteredValue;
+      this.phastCO2SavingsService.modificationCo2SavingsData.next(modificationCo2SavingsData);
+    }
+  }
+
+  setFuelOptions(isUserFormChange = false) {
+    let otherFuelOptions: OtherFuel = _.find(this.otherFuels, (val) => { return this.form.controls.energySource.value === val.energySource; });
+    this.fuelOptions = otherFuelOptions.fuelTypes;
+    let fuelType: string = this.fuelOptions[0].fuelType;
+    if (!isUserFormChange && this.co2SavingsData.fuelType) {
+      // take existing selection
+      fuelType = this.co2SavingsData.fuelType;
+    }
+
+    this.form.patchValue({
+      fuelType: fuelType
+    });
+    this.setFuelTypeOutputRate(isUserFormChange);
+  }
+
+
+  setFuelTypeOutputRate(isUserFormChange = false) {
+    if (this.isBaseline) {
+      this.setBaselineFuelTypeOutputRate(isUserFormChange);
+    } else {
+      this.setModificationFuelTypeOutputRate()
+    }
+    this.calculate();
+  }
+
+  setBaselineFuelTypeOutputRate(isUserFormChange: boolean) {
+    console.log('isUserFuelTypeChange', isUserFormChange)
+    console.log('userEntered baseline', this.form.controls.userEnteredBaselineEmissions.value)
+
+    if (isUserFormChange || !this.form.controls.userEnteredBaselineEmissions.value) {
+      let selectedFuelOutputRate: number = _.find(this.fuelOptions, (val) => { return this.form.controls.fuelType.value === val.fuelType; }).outputRate;
+      if(this.settings.energyResultUnit != 'MMBtu'){
+        selectedFuelOutputRate = this.convertUnitsService.value(selectedFuelOutputRate).from('MMBtu').to(this.settings.energyResultUnit);
+      }
+      
+      this.form.patchValue({
+        totalEmissionOutputRate: selectedFuelOutputRate
+      });
+      this.setUserEnteredModificationEmissions(false);
+    } 
+  }
+
+  setModificationFuelTypeOutputRate() {
+    if (!this.form.controls.userEnteredModificationEmissions.value) {
+      let selectedFuelOutputRate: number = _.find(this.fuelOptions, (val) => { return this.form.controls.fuelType.value === val.fuelType; }).outputRate;
+      if(this.settings.energyResultUnit != 'MMBtu'){
+        selectedFuelOutputRate = this.convertUnitsService.value(selectedFuelOutputRate).from('MMBtu').to(this.settings.energyResultUnit);
+      }
+      this.form.patchValue({
+        totalEmissionOutputRate: selectedFuelOutputRate
+      });
+      this.form.controls.userEnteredModificationEmissions.patchValue(false);
+    }
+  }
+  
+  setSubRegionData() {
+    this.egridService.getAllSubRegions();
```

Comment:
This should probably be called in phast.component (see the pumps implementation). CSV data won't be read in time here - this is why when you create an electrotech assessment with a zipcode it doesn't populate an emissions output rate